### PR TITLE
CakePHP 4.x compatibility

### PR DESCRIPTION
**Pull request by swiffer**
- code style
- type hints

### CODE REVIEW

File: src/TestSuite/Fixture/ChecksumTestFixture.php [hunk 1]
```diff
@@ -15,43 +17,43 @@
  */
 class ChecksumTestFixture extends TestFixture
 {
+    /**
+     * List of table hashes
+     *
+     * @var array
+     */
+    public static $tableHashes = [];
```
Comment (ADmad) [inline src/TestSuite/Fixture/ChecksumTestFixture.php]: Does this property need to be public? Given the underscore prefix earlier, the "public" visibility might be a mistake and it should be "protected" instead.

**Comment by swiffer** [comment]
@ADmad I've added compatibility for Postgres as well (https://stackoverflow.com/a/13948327).

Can you take a look please?

**Comment by ADmad** [comment]
I don't have enough knowledge regarding Postgres to be able to review that. Hopefully someone else can.

**Comment by swiffer** [comment]
I've tested the Statement locally, worked fine here

**Comment by swiffer** [comment]
@ADmad I had to revert the changes for postgres because it does not work when foreign key constraints exists. this is CakePHP 4.0 compatibility only now.

File: src/TestSuite/Fixture/ChecksumTestFixture.php [hunk 2]
```diff
@@ -113,11 +99,13 @@ protected function _hash(ConnectionInterface $db): string
 
         if ($driver instanceof Mysql) {
             $sth = $db->execute("CHECKSUM TABLE " . $this->table . ';');
+
```
Comment (jippi) [inline src/TestSuite/Fixture/ChecksumTestFixture.php:117]: these newlines seem to be unrelated to compat with 4.x

File: src/TestSuite/Fixture/ChecksumTestFixture.php [hunk 3]
```diff
@@ -26,15 +26,9 @@ class ChecksumTestFixture extends TestFixture
     protected static $_tableHashes = [];
 
     /**
```
Comment (jippi) [inline src/TestSuite/Fixture/ChecksumTestFixture.php:27]: these comments are _not_ the same as the parent one - so these should be retained

**Review by jippi — CHANGES_REQUESTED** [review]

File: src/TestSuite/Fixture/ChecksumTestFixture.php (hunk 2 above)
Comment (swiffer) [reply src/TestSuite/Fixture/ChecksumTestFixture.php:117]: I got this error in phpcs:

Missing blank line before return statement

File: src/TestSuite/Fixture/ChecksumTestFixture.php (hunk 3 above)
Comment (swiffer) [reply src/TestSuite/Fixture/ChecksumTestFixture.php:27]: added them back ;)

**Comment by swiffer** [comment]
@jippi can you take a look again so we can get this merged ?

**Comment by dakota** [comment]
What is still needed for this?

**Review by ADmad — APPROVED** [review]
LGTM
